File: core/osrd-railjson/src/main/java/fr/sncf/osrd/railjson/schema/common/ExcludeFromGeneratedCodeCoverage.java
package fr.sncf.osrd.railjson.schema.common;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.*;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Documented
@Retention(RUNTIME)
@Target({CONSTRUCTOR, TYPE, METHOD})
public @interface ExcludeFromGeneratedCodeCoverage {
}

File: core/osrd-railjson/src/main/java/fr/sncf/osrd/railjson/schema/common/ExcludeFromGeneratedCodeCoverage.kt
package fr.sncf.osrd.railjson.schema.common

@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(
    AnnotationTarget.CONSTRUCTOR,
    AnnotationTarget.CLASS,
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
)
annotation class ExcludeFromGeneratedCodeCoverage
